### ISSUE TITLE
xService not updating service configuration

### ISSUE THREAD

**Issue by gigi81** [PowerShell/xPSDesiredStateConfiguration#126]
Once the service has been created, the xService dscresource will not update the service configuration even if one or more service properties are not compliant with the configuration.
For example: depencencies, credentails, displayname, description, etc., even if one or more don't match the current service configuration, they won't be corrected.
The only solution at the moment is to manually delete the service and let DSC recreate them with the correct configuration. But I don''t think it's how it is supposed to work.

**Comment by gigi81** [comment]
To be fair, I had a second look at the code, and some of the properties are actually made consistent even if the service already exists, most importantly for me the credentials.
Anyway, there are still other properties that will not be consistent like dependencies and description.

**Comment by spawluk** [comment]
+1

**Comment by PlagueHO** [comment]
I'll try and look into this over the weekend (or possibly the next) - unless someone else gets to it first.

**Comment by kwirkykat** [comment]
#159 addresses come of these. Though I'm not sure DisplayName and Description will get updated still

**Comment by bswerdlen** [comment]
I have been dealing with this issue on my end as well, primarily when updating the configuration of other resources within the document. It is breaking all of our builds and appears to be causing larger issues on our servers because DSC thinks it's configuration has failed when it has not.

I have patched this on my end to continue our normal business operations. The intended goal of my change was to be able to re-run a successful DSC configuration without producing false errors and to be able to update the LogOnAs account for the service via DSC

The only change I made to get everything functioning properly was the below update to 'Set-TargetResource' 
Lines 240-244
if(CompareServicePath -Path $Path -Name $Name)
        {
            ThrowInvalidArgumentError "ServiceExistsSamePath" ($LocalizedData.ServiceExistsSamePath -f $Name, $Path)
        }
        ThrowInvalidArgumentError "ServiceAlreadyExists" ($LocalizedData.ServiceAlreadyExists -f $Name)

Was changed to
if(-not (CompareServicePath -Path $Path -Name $Name))
        {
            ThrowInvalidArgumentError "ServiceAlreadyExists" ($LocalizedData.ServiceAlreadyExists -f $Name)
        }

I'm not sure why the DSC Resource was programmed to throw an error when a service exists with the same name and path (since that implies we are talking about the same service and just trying to reconfigure it); so I just completely removed that logic. 

Hope this helps!
